catalog/plugin/catalog-plugin-metacardingest-network/src/test/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeHelperTest.java

```diff
@@ -0,0 +1,351 @@
+                metadataResult);
+
+        Map pocResult = ImmutableMap.builder()
+                .putAll(attributeMetadata)
+                .putAll(attributeCreated)
+                .build();
+        assertThatMetacardHasAttributesEqualToThoseInMap(metacardWithPoc, pocResult);
+
+        Map createdResult = ImmutableMap.builder()
+                .putAll(attributeMetadata)
+                .putAll(attributePoc)
+                .build();
+        assertThatMetacardHasAttributesEqualToThoseInMap(metacardWithCreatedDate, createdResult);
+    }
+
+    @Test
+    public void testCreateAttributeInteger() throws Exception {
+        runParameterizedAttributeTest(INTEGER, "186", 186);
+    }
+
+    @Test
+    public void testCreateAttributeDouble() throws Exception {
+        runParameterizedAttributeTest(DOUBLE, "0.8866432", 0.8866432);
+    }
+
+    @Test
+    public void testCreateAttributeFloat() throws Exception {
+        runParameterizedAttributeTest(FLOAT, "0.123", 0.123f);
+    }
+
+    @Test
+    public void testCreateAttributeShort() throws Exception {
+        runParameterizedAttributeTest(SHORT, "16", (short) 16);
+    }
+
+    @Test
+    public void testCreateAttributeLong() throws Exception {
+        runParameterizedAttributeTest(LONG, "123456789", 123456789L);
+    }
+
+    @Test
+    public void testCreateAttributeBoolean() throws Exception {
+        runParameterizedAttributeTest(BOOLEAN, "true", true);
+    }
+
+    @Test
+    public void testCreateAttributeBinary() throws Exception {
+        String binaryString = "0100101010110110010010110110010";
+        runParameterizedAttributeTest(BINARY, binaryString, binaryString.getBytes());
+    }
+
+    @Test
+    public void testCreateAttributeDate() throws Exception {
+        String lexicalXsdDateTime = "2001-10-26T21:32:52";
+        runParameterizedAttributeTest(DATE,
+                lexicalXsdDateTime,
+                DatatypeConverter.parseDateTime(lexicalXsdDateTime)
+                        .getTime());
+    }
+
+    @Test
+    public void testCreateAttributeString() throws Exception {
+        runParameterizedAttributeTest(STRING, SAMPLE_JSON_STRING, SAMPLE_JSON_STRING);
+    }
+
+    @Test
+    public void testCreateAttributeObject() throws Exception {
+        runParameterizedAttributeTest(OBJECT, SAMPLE_JSON_STRING, SAMPLE_JSON_STRING);
+    }
+
+    @Test
+    public void testCreateAttributeGeometry() throws Exception {
+        runParameterizedAttributeTest(GEOMETRY, SAMPLE_JSON_STRING, SAMPLE_JSON_STRING);
+    }
+
+    @Test
+    public void testCreateAttributeXml() throws Exception {
+        runParameterizedAttributeTest(XML, SAMPLE_JSON_STRING, SAMPLE_JSON_STRING);
+    }
+
+    @Test
+    public void testCreateAttributeWithIllegalArgument() throws Exception {
+        haveDescriptorReturnFormatOf(INTEGER);
+        Attribute attribute = attributeHelper.createAttribute(mockDescriptor, "1874xyz");
+        assertNull(attribute);
+    }
+
```

Comment:
Looks like we're not testing the `AttributeHelper()` constructor that takes in a `SortedServiceReferenceList` and `createAttribute` with a multi-valued `parsedValue`.